tests/coreos-install/util/util.go

```diff
@@ -106,14 +115,43 @@ func FetchLocalImage(t *testing.T) string {
 	return tmpDir
 }
 
-func DownloadFile(tmpDir, name string) error {
+// Used to get defaults for channel, board, & version, first checks if the
+// host machine is Container Linux and if so uses the data from the machine
+// otherwise defaults to stable, amd64-usr, & current respectively
+func GetDefaults(t *testing.T) (string, string, string) {
+	data, err := ioutil.ReadFile("/usr/lib/os-release")
+	if err != nil {
+		return "stable", "amd64-usr", "current"
+	}
+
+	os, err := regexpSearch("id", "ID=(.*)", data)
+	if err != nil || os != "coreos" {
+		return "stable", "amd64-usr", "current"
+	}
+
+	version := RegexpSearch(t, "version", "VERSION_ID=(.*)", data)
+	board := RegexpSearch(t, "board", "COREOS_BOARD=\"(.*)\"", data)
+
+	data, err = ioutil.ReadFile("/etc/coreos/update.conf")
+	if err != nil {
+		t.Fatalf("reading /etc/coreos/update.conf: %v", err)
+	}
+
+	channel := RegexpSearch(t, "channel", "GROUP=(.*)", data)
```

Comment:
I don't know if there are any quoting rules for this file, but best not to make any assumptions.